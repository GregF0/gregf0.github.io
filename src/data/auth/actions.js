import { firebase } from '../../firebase/firebase'

export const setUserInfo = (userInfo) => ({
  type: 'SET_USER_INFO',
  userInfo,
})

export const linkAccount = (authProvider) => {
  return async (dispatch) => {
    try {
      const res = await firebase.auth().currentUser.linkWithPopup(authProvider)

      const user = res.user

      const userInfo = { uid: user.uid }

      userInfo.creationTime = user.metadata.creationTime

      if (user.providerData && user.providerData.length) {
        userInfo.name = user.providerData[0].displayName
        userInfo.photo = user.providerData[0].photoURL
      }

      dispatch(setUserInfo(userInfo))
      return res
    } catch (e) {
      if (e.code === 'auth/credential-already-in-use') {
        return firebase.auth().signInWithCredential(e.credential)
      }
    }
  }
}

export const signOut = () => ({
  type: 'SIGN_OUT',
})

export const startSignOut = () => {
  return async (dispatch) => {
    await firebase.auth().signOut()
    dispatch(signOut())
  }
}
