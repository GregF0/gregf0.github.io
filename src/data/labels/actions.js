import fs from '../../firebase/firebase'

export const setFullscreenDialog = (fullscreenDialog) => ({
  type: 'SET_FULLSCREEN_DIALOG',
  fullscreenDialog,
})

export const setDesktopDialog = (desktopDialog) => ({
  type: 'SET_DESKTOP_DIALOG',
  desktopDialog,
})

export const setDeleteAlert = (deleteAlert) => ({
  type: 'SET_DELETE_ALERT',
  deleteAlert,
})

export const setMenuOpened = (menuOpened) => ({
  type: 'SET_MENU_OPENED',
  menuOpened,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})

export const setLabelEditting = (labelEditting) => ({
  type: 'SET_LABEL_EDITTING',
  labelEditting,
})

export const setLabelSelected = (labelSelected) => ({
  type: 'SET_LABEL_SELECTED',
  labelSelected,
})

export const addLabel = (label) => ({
  type: 'ADD_LABEL',
  label,
})

export const startAddLabel = (label) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const newLabelRef = fs.collection(`users/${uid}/labels`).doc()

    dispatch(
      addLabel({
        id: newLabelRef.id,
        ...label,
      })
    )
    await newLabelRef.set({ ...label })
  }
}

export const editLabel = (id, updates) => ({
  type: 'EDIT_LABEL',
  id,
  updates,
})

export const startEditLabel = (id, updates) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(editLabel(id, updates))

    await fs.doc(`users/${uid}/labels/${id}`).update({
      ...updates,
    })
  }
}

export const deleteLabel = (id) => ({
  type: 'DELETE_LABEL',
  id,
})

export const startDeleteLabel = (id) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(deleteLabel(id))

    await fs.doc(`users/${uid}/labels/${id}`).delete()
  }
}

export const setLabels = (labels) => ({
  type: 'SET_LABELS',
  labels,
})

export const startSetLabels = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    try {
      const labelsRef = await fs.collection(`users/${uid}/labels`).get()

      const labels = labelsRef.docs.map((label) => ({
        id: label.id,
        ...label.data(),
      }))

      dispatch(setLabels(labels))

      return labelsRef
    } catch (e) {
      dispatch(setLabels([]))
    }
  }
}
