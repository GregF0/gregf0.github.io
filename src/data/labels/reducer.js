export const initialState = {
  fullscreenDialog: false,
  desktopDialog: false,
  deleteAlert: false,
  menuOpened: false,
  formValue: null,
  labelEditting: null,
  labelSelected: null,
  data: [],
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_FULLSCREEN_DIALOG':
      return {
        ...state,
        fullscreenDialog: action.fullscreenDialog,
      }
    case 'SET_DESKTOP_DIALOG':
      return {
        ...state,
        desktopDialog: action.desktopDialog,
      }
    case 'SET_DELETE_ALERT':
      return {
        ...state,
        deleteAlert: action.deleteAlert,
      }
    case 'SET_MENU_OPENED':
      return {
        ...state,
        menuOpened: action.menuOpened,
      }
    case 'SET_LABEL_EDITTING':
      return {
        ...state,
        labelEditting: action.labelEditting,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    case 'ADD_LABEL':
      return {
        ...state,
        data: [...state.data, action.label],
        labelSelected: state.labelSelected || action.label,
      }
    case 'EDIT_LABEL':
      const updatedLabels = state.data.map((label) => {
        if (label.id === action.id) {
          return {
            ...label,
            ...action.updates,
          }
        } else {
          return label
        }
      })
      return {
        ...state,
        data: updatedLabels,
        labelSelected:
          state.labelSelected && state.labelSelected.id === action.id
            ? { ...state.labelSelected, ...action.updates }
            : state.labelSelected,
      }
    case 'DELETE_LABEL':
      return {
        ...state,
        data: state.data.filter(({ id }) => id !== action.id),
        labelSelected:
          state.labelSelected && state.labelSelected.id === action.id
            ? null
            : state.labelSelected,
      }
    case 'SET_LABELS':
      return {
        ...state,
        data: action.labels,
      }
    case 'SET_LABEL_SELECTED':
      return {
        ...state,
        labelSelected: action.labelSelected,
      }
    default:
      return state
  }
}
