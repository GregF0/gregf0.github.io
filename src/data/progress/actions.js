export const setProgressVisibility = (visibility) => ({
  type: 'SET_PROGRESS_VISIBILITY',
  visibility,
})
