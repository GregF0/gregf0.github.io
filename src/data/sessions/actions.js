import fs from '../../firebase/firebase'

export const setSessions = (sessions) => ({
  type: 'SET_SESSIONS',
  sessions,
})

export const startSetSessions = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    try {
      const sessionsRef = await fs.collection(`users/${uid}/sessions`).get()

      const sessions = sessionsRef.docs.map((session) => ({
        id: session.id,
        ...session.data(),
      }))

      dispatch(setSessions(sessions))
      return sessionsRef
    } catch (e) {
      dispatch(setSessions([]))
    }
  }
}

export const addSession = (session) => ({
  type: 'ADD_SESSION',
  session,
})

export const startAddSession = (session) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid
    const { duration, label = null, createdAt } = session

    const newSessionRef = fs.collection(`users/${uid}/sessions`).doc()

    dispatch(
      addSession({
        id: newSessionRef.id,
        ...session,
      })
    )

    await newSessionRef.set({
      duration,
      label,
      createdAt,
    })

    return newSessionRef
  }
}
