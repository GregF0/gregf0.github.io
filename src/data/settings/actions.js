import fs from '../../firebase/firebase'

export const setWorkDuration = (duration) => ({
  type: 'SET_WORK_DURATION',
  duration,
})

export const startSetWorkDuration = (duration) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setWorkDuration(duration))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { workDuration: duration } }, { merge: true })
  }
}

export const setShortBreakDuration = (duration) => ({
  type: 'SET_SHORT_BREAK_DURATION',
  duration,
})

export const startSetShortBreakDuration = (duration) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setShortBreakDuration(duration))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { shortBreakDuration: duration } }, { merge: true })
  }
}

export const setLongBreakDuration = (duration) => ({
  type: 'SET_LONG_BREAK_DURATION',
  duration,
})

export const startSetLongBreakDuration = (duration) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setLongBreakDuration(duration))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { longBreakDuration: duration } }, { merge: true })
  }
}

export const setRounds = (rounds) => ({
  type: 'SET_ROUNDS',
  rounds,
})

export const startSetRounds = (rounds) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setRounds(rounds))

    await fs.doc(`users/${uid}`).set({ settings: { rounds } }, { merge: true })
  }
}

export const setShowTimerInTitle = (showTimerInTitle) => ({
  type: 'SET_SHOW_TIMER_IN_TITLE',
  showTimerInTitle,
})

export const startSetShowTimerInTitle = (showTimerInTitle) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setShowTimerInTitle(showTimerInTitle))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { showTimerInTitle } }, { merge: true })
  }
}

export const setShowNotifications = (showNotifications) => ({
  type: 'SET_SHOW_NOTIFICATIONS',
  showNotifications,
})

export const startSetShowNotifications = (showNotifications) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setShowNotifications(showNotifications))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { showNotifications } }, { merge: true })
  }
}

export const setDarkMode = (darkMode) => ({
  type: 'SET_DARK_MODE',
  darkMode,
})

export const startSetDarkMode = (darkMode) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    localStorage.setItem('darkMode', +darkMode)

    dispatch(setDarkMode(darkMode))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { darkMode } }, { merge: true })
  }
}

export const setAutostart = (autostart) => ({
  type: 'SET_AUTOSTART',
  autostart,
})

export const startSetAutostart = (autostart) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setAutostart(autostart))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { autostart } }, { merge: true })
  }
}

export const setFirstDayOfTheWeek = (firstDayOfTheWeek) => ({
  type: 'SET_FIRST_DAY_OF_THE_WEEK',
  firstDayOfTheWeek,
})

export const startSetFirstDayOfTheWeek = (firstDayOfTheWeek) => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    dispatch(setFirstDayOfTheWeek(firstDayOfTheWeek))

    await fs
      .doc(`users/${uid}`)
      .set({ settings: { firstDayOfTheWeek } }, { merge: true })
  }
}

export const setSettings = (settings) => ({
  type: 'SET_SETTINGS',
  settings,
})

export const startSetSettings = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    try {
      const docRef = await fs.doc(`users/${uid}`).get()
      let data = {}

      if (docRef.exists) {
        data = docRef.data().settings
      }

      dispatch(setSettings(data))
    } catch (e) {
      dispatch(setSettings({}))
    }
  }
}
