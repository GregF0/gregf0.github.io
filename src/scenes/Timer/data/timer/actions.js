export const setStatus = (status) => ({
  type: 'SET_STATUS',
  status,
})

export const setProgress = (progress) => ({
  type: 'SET_PROGRESS',
  progress,
})

export const setTimeLeft = (timeLeft) => ({
  type: 'SET_TIME_LEFT',
  timeLeft,
})

export const saveInterval = (interval) => ({
  type: 'SAVE_INTERVAL',
  interval,
})

export const pauseTimer = () => ({
  type: 'PAUSE_TIMER',
})

export const resetTimer = ({ duration, showTimerInTitle }) => ({
  type: 'RESET_TIMER',
  duration,
  showTimerInTitle,
})

export const setNextTimer = (settings) => ({
  type: 'SET_NEXT_TIMER',
  settings,
})

export const setSaveSessionAlert = (saveSessionAlert) => ({
  type: 'SET_SAVE_SESSION_ALERT',
  saveSessionAlert,
})
