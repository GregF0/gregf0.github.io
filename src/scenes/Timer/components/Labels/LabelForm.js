import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import TextField from '@material-ui/core/TextField'
import MenuItem from '@material-ui/core/MenuItem'
import styled, { css } from 'styled-components'
import colors from './Fixtures/colors'
import { setFormValue } from '../../../../data/labels/actions'
import { useTheme } from '@material-ui/core'

export const LabelForm = () => {
  const { labelEditting } = useSelector((state) => state.labels)
  const darkMode = useSelector((state) => +state.settings.darkMode)

  const [name, setName] = useState(labelEditting ? labelEditting.name : '')
  const [nameValid, setNameValid] = useState(true)
  const nameHelperText = 'Fill in label name'

  const [color, setColor] = useState(labelEditting ? labelEditting.color : '')

  const dispatch = useDispatch()

  const onNameChange = (e) => {
    const value = e.target.value

    setName(value)
    dispatch(setFormValue({ name: value, color }))

    if (!!value !== nameValid) {
      setNameValid(!nameValid)
    }
  }

  const onColorChange = (e) => {
    const value = e.target.value

    setColor(value)
    dispatch(setFormValue({ name, color: value }))
  }

  const onNameBlur = () => {
    if (!name) {
      setNameValid(false)
    }
  }

  const theme = useTheme()

  return (
    <Form
      noValidate
      autoComplete="off"
      darkMode={darkMode}
      color={theme.palette.primary.light}
    >
      <TextField
        error={!nameValid}
        helperText={!nameValid ? nameHelperText : ' '}
        required
        id="label-name"
        label="Label name"
        variant="filled"
        name="name"
        margin="normal"
        value={name}
        onChange={onNameChange}
        onBlur={onNameBlur}
        fullWidth
        autoFocus
      />

      <ColorTextField
        select
        required
        id="label-color"
        label="Label color"
        variant="filled"
        name="color"
        margin="normal"
        value={color}
        onChange={onColorChange}
        fullWidth
      >
        {colors.map((color) => (
          <MenuItem key={color.id} value={color.id}>
            <ColorIndicator
              color={darkMode ? color.hex.darkMode : color.hex.normal}
            />
            <span>{color.name}</span>
          </MenuItem>
        ))}
      </ColorTextField>
    </Form>
  )
}

const Form = styled.form`
  ${({ darkMode, color }) =>
    darkMode &&
    css`
      & label.Mui-focused:not(.Mui-error) {
        color: ${color};
      }

      & .MuiFilledInput-underline.Mui-focused:not(.Mui-error):after {
        border-bottom-color: ${color};
      }
    `}
`

const ColorTextField = styled(TextField)`
  #label-color {
    display: flex;
    align-items: center;
  }
`

const ColorIndicator = styled.span`
  background: ${({ color }) => color};
  display: inline-block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  margin-right: 20px;
`

export { ColorTextField }
