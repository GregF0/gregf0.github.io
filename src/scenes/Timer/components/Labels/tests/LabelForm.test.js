import React from 'react'
import * as redux from 'react-redux'
import { createShallow } from '@material-ui/core/test-utils'
import TextField from '@material-ui/core/TextField'
import labels from '../../../../../data/labels/tests/mock-data/labels'
import { LabelForm, ColorTextField } from '../LabelForm'
import * as labelsActions from '../../../../../data/labels/actions'

describe('<LabelForm />', () => {
  const shallow = createShallow()
  const createWrapper = () => {
    return shallow(<LabelForm />)
  }

  const dispatchMocked = jest.fn()
  jest.spyOn(redux, 'useDispatch').mockImplementation(() => dispatchMocked)

  const createStore = (labelEditting = null) => {
    const store = {
      labels: { labelEditting },
      settings: { darkMode: false },
    }

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((callback) => callback(store))

    return store
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render <LabelForm /> correctly', () => {
    createStore()
    expect(createWrapper()).toMatchSnapshot()
  })

  test('should display error message if name field is invalid', () => {
    const labelEditting = labels[0]
    createStore(labelEditting)

    const wrapper = createWrapper()

    wrapper
      .find(TextField)
      .at(0)
      .simulate('change', { target: { value: '' } })

    expect(wrapper.find(TextField).at(0).prop('error')).toBe(true)
  })

  test('should call setFormValue and should not display error message if name field is valid', () => {
    const setFormValueMocked = jest
      .spyOn(labelsActions, 'setFormValue')
      .mockImplementation(() => jest.fn())

    const labelEditting = labels[0]
    createStore(labelEditting)

    const wrapper = createWrapper()

    wrapper
      .find(TextField)
      .at(0)
      .simulate('change', { target: { value: `${labelEditting.name} -` } })

    expect(wrapper.find(TextField).at(0).prop('error')).toBe(false)

    expect(setFormValueMocked).toHaveBeenCalledWith({
      name: `${labelEditting.name} -`,
      color: labelEditting.color,
    })
  })

  test('should display error message on blur if name field is empty', () => {
    createStore()

    const wrapper = createWrapper()

    wrapper
      .find(TextField)
      .at(0)
      .simulate('blur', { target: { value: '' } })

    expect(wrapper.find(TextField).at(0).prop('error')).toBe(true)
  })

  test('should call setFormValue on color value change', () => {
    const setFormValueMocked = jest
      .spyOn(labelsActions, 'setFormValue')
      .mockImplementation(() => jest.fn())

    const labelEditting = labels[0]
    createStore(labelEditting)

    const wrapper = createWrapper()

    wrapper.find(ColorTextField).prop('onChange')({
      target: { value: '#ffeb3b' },
    })

    expect(setFormValueMocked).toHaveBeenCalledWith({
      name: labelEditting.name,
      color: '#ffeb3b',
    })
  })
})
