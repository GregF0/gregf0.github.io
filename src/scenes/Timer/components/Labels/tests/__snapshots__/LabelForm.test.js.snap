// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelForm /> should render <LabelForm /> correctly 1`] = `
<styled.form
  autoComplete="off"
  color="#7986cb"
  darkMode={0}
  noValidate={true}
>
  <WithStyles(ForwardRef(TextField))
    autoFocus={true}
    error={false}
    fullWidth={true}
    helperText=" "
    id="label-name"
    label="Label name"
    margin="normal"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    required={true}
    value=""
    variant="filled"
  />
  <Styled(WithStyles(ForwardRef(TextField)))
    fullWidth={true}
    id="label-color"
    label="Label color"
    margin="normal"
    name="color"
    onChange={[Function]}
    required={true}
    select={true}
    value=""
    variant="filled"
  >
    <WithStyles(ForwardRef(MenuItem))
      key="red"
      value="red"
    >
      <styled.span
        color="#f44336"
      />
      <span>
        Red
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="pink"
      value="pink"
    >
      <styled.span
        color="#e91e63"
      />
      <span>
        Pink
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="purple"
      value="purple"
    >
      <styled.span
        color="#9c27b0"
      />
      <span>
        Purple
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="deepPurple"
      value="deepPurple"
    >
      <styled.span
        color="#673ab7"
      />
      <span>
        Deep Purple
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="indigo"
      value="indigo"
    >
      <styled.span
        color="#3f51b5"
      />
      <span>
        Indigo
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="blue"
      value="blue"
    >
      <styled.span
        color="#2196f3"
      />
      <span>
        Blue
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="lightBlue"
      value="lightBlue"
    >
      <styled.span
        color="#03a9f4"
      />
      <span>
        Light Blue
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="cyan"
      value="cyan"
    >
      <styled.span
        color="#00bcd4"
      />
      <span>
        Cyan
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="teal"
      value="teal"
    >
      <styled.span
        color="#009688"
      />
      <span>
        Teal
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="green"
      value="green"
    >
      <styled.span
        color="#4caf50"
      />
      <span>
        Green
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="lightGreen"
      value="lightGreen"
    >
      <styled.span
        color="#8bc34a"
      />
      <span>
        Light Green
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="lime"
      value="lime"
    >
      <styled.span
        color="#cddc39"
      />
      <span>
        Lime
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="yellow"
      value="yellow"
    >
      <styled.span
        color="#ffeb3b"
      />
      <span>
        Yellow
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="amber"
      value="amber"
    >
      <styled.span
        color="#ffc107"
      />
      <span>
        Amber
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="orange"
      value="orange"
    >
      <styled.span
        color="#ff9800"
      />
      <span>
        Orange
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="deepOrange"
      value="deepOrange"
    >
      <styled.span
        color="#ff5722"
      />
      <span>
        Deep Orange
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="brown"
      value="brown"
    >
      <styled.span
        color="#795548"
      />
      <span>
        Brown
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="grey"
      value="grey"
    >
      <styled.span
        color="#9e9e9e"
      />
      <span>
        Grey
      </span>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="blueGrey"
      value="blueGrey"
    >
      <styled.span
        color="#607d8b"
      />
      <span>
        Blue Gray
      </span>
    </WithStyles(ForwardRef(MenuItem))>
  </Styled(WithStyles(ForwardRef(TextField)))>
</styled.form>
`;
